components/datagrid/DataGrid.js

```diff
@@ -0,0 +1,97 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Box from '../box';
+import Cell from './Cell';
+import HeaderCell from './HeaderCell';
+import isComponentOfType from '../utils/is-component-of-type';
+import HeaderRow from './HeaderRow';
+import Row from './Row';
+import cx from 'classnames';
+import theme from './theme.css';
+
+class DataGrid extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    selectable: PropTypes.bool,
+  };
+
+  constructor() {
+    super(...arguments);
+
+    this.handleRowSelectionChange = ::this.handleRowSelectionChange;
+    this.handleHeaderRowSelectionChange = ::this.handleHeaderRowSelectionChange;
+
+    this.state = {
+      selectedRows: [],
+    };
+  }
+
+  handleHeaderRowSelectionChange(value) {
+    const allChildrenIndexes = React.Children.map(this.props.children, child => child.key);
```

Comment:
you're actually getting the `HeaderRow` key with this as well (which is undefined I guess in this case). It's probably never going to give us issues, but it feels a bit wrong.

So either we introduce extra levels into the datagrid (`Datagrid.Header` and `Datagrid.Body`?) or you can probably do a `isComponentOfType` check.